src/lib/pages/backoffice/Loan/LoanDetails/LoanActions/ExtendModal/ExtendModal.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'semantic-ui-react';
+import { ConfirmModal } from '@components/ConfirmModal';
+import { withCancel } from '@api/utils';
+import { documentApi } from '@api/documents';
+
+export default class ExtendModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      overbooked: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchCirculation();
+  }
+
+  componentWillUnmount() {
+    this.cancellableCirculation && this.cancellableCirculation.cancel();
+  }
+
+  fetchCirculation = async () => {
+    const { loanDetails, sendErrorNotification } = this.props;
+    const { document_pid } = loanDetails.metadata;
+    try {
+      this.cancellableCirculation = withCancel(documentApi.get(document_pid));
+      const response = await this.cancellableCirculation.promise;
+      const circulation = response.data.metadata.circulation;
+      this.setState({ overbooked: circulation.overbooked === true });
+    } catch (error) {
+      if (error !== 'UNMOUNTED') {
+        sendErrorNotification(
+          'We could not determine if the literature is overbooked.',
+          'Reload the page or contact the support if the problem persists.'
+        );
+      }
+    }
+  };
+
+  render() {
+    const { overbooked } = this.state;
+    const { pid, loanAction, isLoading } = this.props;
+    return (
```

Comment:
I am not sure this is correct: I think that before rendering any button, the fetch of the `document` should complete.
If I am very fast, I open the modal and click on the extend button without evening knowing/seeing if it was overbooked.